### DISCUSSION TITLE
Using Radiobuttons in a TreeTable

### DISCUSSION THREAD

**Discussion by RenzoDaal** [primefaces/community#899] [PrimeVue] [answered]
I am fairly new to using PrimeVue but I was wondering if it is possible to use Radiobuttons in a TreeTable. I can use the "single" selectionMode, which functions like I want it to, but instead of it highlighting the whole row, I want it to just be a radiobutton at the start of the row. Is this possible?

**Comment by tugcekucukoglu** [comment, marked as answer]
Since TreeTable uses hierarchical data, components that allow only one selection, such as RadioButton, are not preferred in TreeTable.

**Reply by RenzoDaal** [reply]
Thanks for the quick reply! Do you mean not preferred or not possible? Doesn't the "single" selectionMode also only allow for one selection? Could I maybe use checkboxes instead where you can only select one item at a time?

**Reply by RenzoDaal** [reply]
In my case I have already filtered out the rows that have children by giving them a property called "selectable" which is set to false. So in my TreeTable you can only select rows that don't have more children rows.

**Comment by tugcekucukoglu** [comment]
I mean not preferred. TreeTable already supports single selection but without using RadioButton. The selection made with RadioButton will not be suitable for components that use hierarchical data such as TreeTable.
